juju/juju
# 2.6 - Watch Unit Config Changes Via Model Cache

## Description of change

This patch changes the uniter API server `WatchConfigSettingsHash` to watch the model cache instead of state.

## QA steps

- Bootstrap
- `juju deploy redis`.
- Run `juju debug-log --include redis/0` in a separate terminal and observe that the config-changed hook fires for each of the configuration change actions below.
- `juju config redis password=new-pass`.
- `juju ssh 0 cat /etc/redis/redis-charm.conf` and check that the line "requirepass new-pass" is present.
- `juju config redis databases=4`.
- `juju ssh 0 cat /etc/redis/redis-charm.conf` and check that the line "databases 4" is present.
- `export JUJU_DEV_FEATURE_FLAGS=generations`.
- `juju add-branch test-branch`.
- `juju track test-branch redis/0`.
- `juju config redis password=branch-pass --branch test-branch`.
- `juju ssh 0 cat /etc/redis/redis-charm.conf` and check that the line "requirepass branch-pass" is present.
- `juju config redis databases=8 --branch test-branch`.
- `juju ssh 0 cat /etc/redis/redis-charm.conf` and check that the line "databases 8" is present;

## Documentation changes

None.

## Bug reference

N/A
